_common/qiskit/execute.py

```diff
@@ -938,15 +871,17 @@ def transpile_for_metrics(qc):
 # DEVNOTE: this approach does not permit passing of untranspiled circuit through
 # DEVNOTE: currently this only caches a single circuit
 def transpile_and_bind_circuit(circuit, params, backend, basis_gates=None,
-                optimization_level=None, layout_method=None, routing_method=None):
+                optimization_level=None, layout_method=None, routing_method=None,
+                seed_transpiler=None):
                 
     logger.info('transpile_and_bind_circuit()')
     st = time.time()
         
     if do_transpile_for_execute:
         logger.info('transpiling for execute')
         trans_qc = transpile(circuit, backend, basis_gates=basis_gates,
-                optimization_level=optimization_level, layout_method=layout_method, routing_method=routing_method) 
+                optimization_level=optimization_level, layout_method=layout_method, routing_method=routing_method,
+                seed_transpiler=seed_transpiler)
```

Comment:
Provide a random seed using the `seed_transpiler` argument in `transpile()` for reproducibility since the routing step in the transpilation can produce stochastic results.